File: backend/apps/hospital/emergency.py
from rest_framework.decorators import api_view, permission_classes
from rest_framework.permissions import AllowAny
from rest_framework.response import Response
from rest_framework import status


@api_view(['POST'])
@permission_classes([AllowAny])
def emergency_sos(request):
    """Handle emergency SOS requests"""
    latitude = request.data.get('latitude')
    longitude = request.data.get('longitude')
    phone = request.data.get('phone', 'Unknown')
    
    if not latitude or not longitude:
        return Response(
            {'error': 'Location required'},
            status=status.HTTP_400_BAD_REQUEST
        )
    
    # Log emergency request
    import logging
    logger = logging.getLogger('apps')
    logger.critical(
        f"EMERGENCY SOS: Location ({latitude}, {longitude}), Phone: {phone}"
    )
    
    # In production, integrate with:
    # - Emergency services API
    # - SMS gateway
    # - Hospital notification system
    
    return Response({
        'success': True,
        'message': 'Emergency services notified',
        'ambulance_eta': '5-10 minutes',
        'emergency_number': '911'
    })

File: frontend/src/api/emergency.js
import axios from 'axios';

const API_URL = import.meta.env.VITE_API_URL || 'http://localhost:8000/api';

export const sendEmergencySOS = async (latitude, longitude, phone = null) => {
  const response = await axios.post(`${API_URL}/hospital/sos/`, {
    latitude,
    longitude,
    phone,
  });
  return response.data;
};
